### ISSUE TITLE
[Chore] Simplify the generation of the type definition files

### ISSUE THREAD

**Issue by nmn** [facebook/react-strict-dom#223]
React Strict DOM has a single file for Flow type definitions and a single file for TS type definitions.

The build step for this is fairly complex, however:
- We transform the entire folder of source files to individual `.js.flow` and `.d.ts` files using `flow-api-translator`.
- We then use a script to concatenate all the type files into a single file
- We then delete the folder

The script that concatenates the types from all the files is brittle and can break if we accidentally use the same type name in different files.

---

We should be able to simplify this process by using this [Rollup-like bundler that understands Flow syntax](https://github.com/facebook/react-forget/tree/5a4e3a1b7c5bb0679574f20429e9c6e54f5e1068/forest/scripts/flow-bundler).

We will need to wait for this bundler to be migrated to the `hermes` repo and open-sourced as a standalone package.

**Comment by necolas** [comment]
> React Strict DOM has a single file for Flow type definitions and a single file for TS type definitions.

I don't think that has been the case for a while. But it would be good to go back to 1-file libdefs using the flow bundler at some point
